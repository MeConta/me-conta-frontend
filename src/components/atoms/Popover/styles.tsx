import styled from 'styled-components'

export const Container = styled.div`
  position: absolute;
  top: 0;
  left: 100%;
  width: 15rem;
  border-radius: 1rem;
  margin-left: 2rem;
  border: 1px solid ${(p) => p.theme.colors.silverChalice};
  background-color: ${(p) => p.theme.colors.white};
  box-shadow: 0px 2px 4px rgba(0, 0, 0, 0.12);
  z-index: 5;
  & > ::before {
    content: '';
    width: 16px;
    height: 16px;
    background-color: white;
    border-left: 1px solid ${(p) => p.theme.colors.silverChalice};
    border-bottom: 1px solid ${(p) => p.theme.colors.silverChalice};
    border-bottom-left-radius: 4px;
    transform: rotate(45deg);
    position: absolute;
    display: block;
    top: 28%;
    left: -19px;
    z-index: -1;
  }
  @media (max-width: 768px) {
    top: 100%;
    left: unset;
    right: -14px;
    margin-top: 2rem;
    margin-left: 0;
    & > ::before {
      border-left: 1px solid ${(p) => p.theme.colors.silverChalice};
      border-top: 1px solid ${(p) => p.theme.colors.silverChalice};
      border-bottom: none;
      border-top-left-radius: 4px;
      border-bottom-left-radius: 0;
      top: -19px;
      left: 70%;
    }
  }
`

export const TextContainer = styled.div`
  position: relative;
  margin: 1rem;
  background-color: white;
  color: ${(p) => p.theme.colors.mineShaft};
  z-index: 5;
  & > ul {
    padding-top: 4px;
    list-style: inside;
  }
`

export const ListTitle = styled.p`
  font-size: ${(p) => p.theme.font.sizes['desk-medium']};
  font-weight: ${(p) => p.theme.font.semibold};
`
