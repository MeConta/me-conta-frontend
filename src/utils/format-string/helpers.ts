// Rever tipagem
export const formatPhoneNumber = (value: string): any =>
  value
    .replace(/\D/g, '')
    .replace(/(\d{2})(\d)/, '($1) $2')
    .replace(/(\s\d{1})/, '$1 ')
    .replace(/(\d{4})(\d)/, '$1-$2')
    .replace(/(-\d{4})(\d+?)$/, '$1')

export const removePhoneMask = (value: string) => value.replace(/\D/g, '')
