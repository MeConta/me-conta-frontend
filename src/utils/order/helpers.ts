import { VolunteerAvailableSlot } from '../../services/volunteers-service/volunteer-service'

function convertSlotsIntoString(dateSlot: VolunteerAvailableSlot) {
  return new Date(dateSlot.inicio).toLocaleTimeString([], {
    hour: '2-digit',
    minute: '2-digit',
    hour12: false
  })
}

export const sortSlotsByTime = (dateSlot: VolunteerAvailableSlot[]) => {
  return dateSlot.sort((a, b) => {
    const x = convertSlotsIntoString(a)
    const y = convertSlotsIntoString(b)
    return x == y ? 0 : x > y ? 1 : -1
  })
}
