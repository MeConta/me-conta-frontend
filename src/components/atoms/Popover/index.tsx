import * as S from './styles'

export interface PopoverProps {
  title: string
  items: Array<string>
}

export default function Popover(props: PopoverProps) {
  const itemsList = props.items.map((item, index) => (
    <li key={index}>{item}</li>
  ))
  return (
    <S.Container id="popover" data-testid="popoverContainer">
      <S.TextContainer data-testid="popoverText">
        <S.ListTitle data-testid="popoverTextTitle">{props.title}</S.ListTitle>
        <ul data-testid="popoverTextItems">{itemsList}</ul>
      </S.TextContainer>
    </S.Container>
  )
}
