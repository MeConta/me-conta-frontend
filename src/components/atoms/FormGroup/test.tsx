import userEvent from '@testing-library/user-event'

import { act, render, screen } from 'utils/tests/helpers'

import { FormGroup } from '.'
import Popover from '../Popover'

describe('<FormGroup/>', () => {
  it('should render the input with a label, when provided', () => {
    render(
      <FormGroup label="qualquer-label" name="nome">
        <input id="nome" />
      </FormGroup>
    )
    expect(screen.getByLabelText('qualquer-label')).toBeInTheDocument()
  })

  it('should render error input', () => {
    const errorMessage = 'Error message'
    const { container } = render(
      <FormGroup label="qualquer-label" name="nome" error={errorMessage}>
        <input id="nome" />
      </FormGroup>
    )
    expect(screen.getByText(errorMessage)).toBeInTheDocument()
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render popover when it is set', () => {
    render(
      <FormGroup
        label="accessible"
        name="accessible"
        error="accessible"
        popover={<Popover title="Test title" items={['Item 1', 'Item 2']} />}
        showPopover
      >
        <input id="nome" data-testid="inputField" />
      </FormGroup>
    )
    act(() => userEvent.click(screen.getByTestId('inputField')))
    expect(screen.getByTestId('popoverContainer')).toBeInTheDocument()
  })

  it('should render extraContent when it is set', () => {
    render(
      <FormGroup
        label="accessible"
        name="accessible"
        extraContent={<p>10/200</p>}
        showPopover
      >
        <input id="nome" data-testid="inputField" />
      </FormGroup>
    )
    expect(screen.getByText('10/200')).toBeInTheDocument()
  })

  // it('should accessible by tab', () => {
  //   render(<FormGroup label="accessible" name="accessible" />)
  //   const input = screen.getByLabelText('accessible')
  //   expect(document.body).toHaveFocus()
  //   userEvent.tab()
  //   expect(input).toHaveFocus()
  // })

  // it('should accessible by tab when disabled', () => {
  //   render(<FormGroup label="accessible" name="accessible" disabled />)
  //   const input = screen.getByLabelText('accessible')
  //   expect(document.body).toHaveFocus()
  //   userEvent.tab()
  //   expect(input).not.toHaveFocus()
  // })
})
