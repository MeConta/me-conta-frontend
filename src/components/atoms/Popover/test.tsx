import theme from 'styles/theme'
import { render, screen } from 'utils/tests/helpers'
import Popover from '.'

describe('<Popover/>', () => {
  it('should render popover', () => {
    render(<Popover title="Test title" items={['Item 1', 'Item 2']} />)
    expect(screen.getByText('Test title')).toBeInTheDocument()
    expect(screen.getByText('Item 1')).toBeInTheDocument()
    expect(screen.getByText('Item 2')).toBeInTheDocument()
  })

  it('should render popover styles correctly', () => {
    render(<Popover title="Test title" items={['Item 1', 'Item 2']} />)
    expect(screen.getByTestId('popoverContainer')).toHaveStyle({
      position: 'absolute',
      top: '0',
      left: '100%',
      width: '15rem',
      borderRadius: '1rem',
      marginLeft: '2rem'
    })
    expect(screen.getByTestId('popoverText')).toHaveStyle({
      position: 'relative',
      margin: '1rem',
      backgroundColor: 'white',
      color: theme.colors.mineShaft
    })
    expect(screen.getByTestId('popoverTextTitle')).toHaveStyle({
      fontSize: theme.font.sizes['desk-medium'],
      fontWeight: theme.font.semibold
    })
    expect(screen.getByTestId('popoverTextItems')).toHaveStyle({
      paddingTop: '4px',
      listStyle: 'inside'
    })
  })
})
