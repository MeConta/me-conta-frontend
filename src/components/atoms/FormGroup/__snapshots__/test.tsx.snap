// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormGroup/> should render error input 1`] = `
.c0 {
  position: relative;
  margin-bottom: 1.6rem;
}

.c0 .c3 {
  border-color: #DE3163;
}

.c0 .c1 {
  color: #DE3163;
}

.c0 > #popover {
  -webkit-animation-name: fHbZyC;
  animation-name: fHbZyC;
  -webkit-animation-duration: 0.3s;
  animation-duration: 0.3s;
  -webkit-animation-fill-mode: forwards;
  animation-fill-mode: forwards;
  -webkit-animation-timing-function: ease-out;
  animation-timing-function: ease-out;
}

.c2 {
  font-size: 1.5rem;
  font-weight: 700;
  color: #848A8C;
}

.c2[aria-required='true']:after {
  content: ' *';
  color: #DE3163;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 0.7rem;
  padding: 0.8rem 1.6rem;
  margin-top: 1.13rem;
  border: 0.1rem solid;
  border-color: #848A8C;
}

.c4:focus-within {
  box-shadow: 0 0 0.5rem #0545AB;
}

.c5 {
  color: #DE3163;
  display: block;
  font-size: 1rem;
  margin-top: 1rem;
}

<div
  class="c0"
>
  <label
    class="c1 c2"
    data-testid="qualquer-label"
    for="nome"
  >
    qualquer-label
  </label>
  <div
    class="c3 c4"
  >
    <input
      id="nome"
    />
  </div>
  <p
    class="c5"
  >
     
    Error message
     
  </p>
</div>
`;
